consensus/enclave/impl/src/lib.rs

```diff
@@ -2340,6 +2341,330 @@ mod tests {
+            12,
+            &recipient1.default_subaddress(),
+            &mut rng,
+        );
+
+        let enclave = SgxConsensusEnclave::new(logger.clone());
+        let blockchain_config = BlockchainConfig {
+            block_version,
+            governors_map: governors_map.clone(),
+            governors_signature: sign_governors_map(&governors_map),
+            ..Default::default()
+        };
+        enclave
+            .enclave_init(
+                &Default::default(),
+                &Default::default(),
+                &None,
+                blockchain_config,
+            )
+            .unwrap();
+
+        // Initialize a ledger.
+        let sender = AccountKey::random(&mut rng);
+        let mut ledger = create_ledger();
+
+        // We want the next block that gets appended to the ledger to exceed the
+        // tombstone limit of the mint config tx, since we want to make sure
+        // that minting that relies on an old MintConfigTx (one that is past
+        // its tombstone block) still validate and mint successfully.
+        let n_blocks = mint_config_tx.prefix.tombstone_block;
+        initialize_ledger(block_version, &mut ledger, n_blocks, &sender, &mut rng);
+
+        // Form block
+        let parent_block = ledger.get_block(ledger.num_blocks().unwrap() - 1).unwrap();
+
+        let root_element = ledger.get_root_tx_out_membership_element().unwrap();
+
+        let (block, block_contents, signature) = enclave
+            .form_block(
+                &parent_block,
+                FormBlockInputs {
+                    mint_txs_with_config: vec![(
+                        mint_tx.clone(),
+                        mint_config_tx.clone(),
+                        mint_config_tx.prefix.configs[0].clone(),
+                    )],
+                    ..Default::default()
+                },
+                &root_element,
+            )
+            .unwrap();
+
+        // Verify signature
+        assert_eq!(
+            signature.signer(),
+            &enclave
+                .ake
+                .get_identity()
+                .signing_keypair
+                .lock()
+                .unwrap()
+                .public_key()
+        );
+
+        assert!(signature.verify(&block).is_ok());
+
+        // The block contents should contain two mint txs.
+        assert_eq!(block_contents.mint_txs, vec![mint_tx]);
+
+        // There should be no key images or mint config txs
+        assert!(block_contents.key_images.is_empty());
+        assert!(block_contents.validated_mint_config_txs.is_empty());
+
+        // The block contents should contain the minted tx out.
+        assert_eq!(block_contents.outputs.len(), 1);
+
+        let output1 = &block_contents.outputs[0];
+        let (amount, _) = output1
+            .view_key_match(recipient1.view_private_key())
+            .unwrap();
+        assert_eq!(amount.value, 12);
+        assert_eq!(amount.token_id, token_id1);
```

Comment:
:upside_down_face: take or leave as one sees fit, The main benefits I see to comparing structs:
- The failure output provides a bit more context with the rest of the struct
- It tends to make the tests more concise
- If new members get added to the struct they automatically get compared (this can be a bad thing at times)
```suggestion
        assert_eq!(amount, Amount{value: 12, token_id: token_id1});
```

Reply:
Sounds good, although I'll use `Amount::new()`